smithy-cli/src/main/java/software/amazon/smithy/cli/commands/DependencyUtils.java

```diff
@@ -0,0 +1,118 @@
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+import software.amazon.smithy.build.model.MavenConfig;
+import software.amazon.smithy.build.model.MavenRepository;
+import software.amazon.smithy.build.model.SmithyBuildConfig;
+import software.amazon.smithy.cli.EnvironmentVariable;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.utils.IoUtils;
+
+
+final class DependencyUtils {
+    private static final Logger LOGGER = Logger.getLogger(DependencyUtils.class.getName());
+    private static final File LOCK_FILE = new File("smithy-lock.json");
+    private static final MavenRepository CENTRAL = MavenRepository.builder()
+            .url("https://repo.maven.apache.org/maven2")
+            .build();
+
+    private DependencyUtils() {
+        // Utility Class should not be instantiated
+    }
+
+    /**
+     * Gets the list of Maven Repos configured via smithy-build config and environment variables.
+     * <p>
+     * Environment variables take precedence over config file entries. If no repositories are
+     * defined in either environment variables or config file entries, then the returned set
+     * defaults to a singleton set of just the Maven Central repo.
+     *
+     * @param config Smithy build config to check for maven repository values.
+     * @return set of configured repositories.
+     */
+    public static Set<MavenRepository> getConfiguredMavenRepos(SmithyBuildConfig config) {
+        Set<MavenRepository> repositories = new LinkedHashSet<>();
+
+        String envRepos = EnvironmentVariable.SMITHY_MAVEN_REPOS.get();
+        if (envRepos != null) {
+            for (String repo : envRepos.split("\\|")) {
+                repositories.add(MavenRepository.builder().url(repo.trim()).build());
+            }
+        }
+
+        Set<MavenRepository> configuredRepos = config.getMaven()
+                .map(MavenConfig::getRepositories)
+                .orElse(Collections.emptySet());
+
+        if (!configuredRepos.isEmpty()) {
+            repositories.addAll(configuredRepos);
+        } else if (envRepos == null) {
+            LOGGER.finest(() -> String.format("maven.repositories is not defined in smithy-build.json and the %s "
+                            + "environment variable is not set. Defaulting to Maven Central.",
+                    EnvironmentVariable.SMITHY_MAVEN_REPOS));
+            repositories.add(CENTRAL);
+        }
+
+        return repositories;
+    }
+
+    /**
+     * Computes the combined hash of the artifact set and repositories set.
+     *
+     * @param artifacts set of requested artifact coordinates
+     * @param repositories set of repositories used for dependency resolution
+     * @return combined hash
+     */
+    public static int configHash(Set<String> artifacts, Set<MavenRepository> repositories) {
+        Set<String> hashingSet = new LinkedHashSet<>(artifacts);
```

Comment:
I don't think you actually want to sort these values before hashing. Changing the order of repositories can change the resolved dependencies. You could just create a String[] based on the size of both artifacts and repositories, then use Objects.hash on the array. This will maintain order which matters here.